src/OrleansProviders/Streams/Common/PooledCache/PooledQueueCache.cs

```diff
@@ -65,14 +68,15 @@ public class PooledQueueCache<TQueueMessage, TCachedMessage>
         /// For ordered reliable queues we shouldn't need to notify on every purged event, only on the last event 
         ///   of every set of events that get purged.
         /// </summary>
-        public Action<TCachedMessage> OnPurged { get; set; }
+        public Action<TCachedMessage?,TCachedMessage?> OnPurged { get; set; }
```

Comment:
maybe an update to the comments to understand what are the 2 different messages?